### DISCUSSION TITLE
Jwt current user

### DISCUSSION THREAD

**Discussion by ssahinh** [jasontaylordev/CleanArchitecture#586] [Q&A]
How should I pass user from controller to command? Is there any approach you can recommend?

Im using this approach;

Controller;

string? email = User.FindFirst(ClaimTypes.Email)?.Value;
command.Email = email;
var response = await Mediator.Send(command);

Command;

public class Command : IRequest<bool>
{
    [JsonIgnore]
    public string? Email { get; set; }
}

**Comment by suicodave** [comment]
You can modify the ICurrentUserService and add an Email property just like the UserId and then thru DI you can inject that interface to your CommandHandler and get the user email so you will no longer have to define Email property in your Command

**Reply by ssahinh** [reply]
> You can modify the ICurrentUserService and add an Email property just like the UserId and then thru DI you can inject that interface to your CommandHandler and get the user email so you will no longer have to define Email property in your Command

first of all, thanks for answer.

actually, im not using service and repository pattern. accessing command directly from controller.

**Reply by suicodave** [reply]
Why not?

**Reply by ssahinh** [reply]
I am using entity framework. it already does the necessary abstractions. It doesn't make sense to abstract something that has been abstracted again.

**Reply by suicodave** [reply]
It doesn't make sense to me why It has something to do with the abstraction. ICurrentService is there for a reason and probably all of your handlers have a dependency on IApplicationDBContext so why not consider having an ICurrentService as a dependency?. Consider the following:

```cs

    public class NewEmployeeCommand:IRequest<int>
    {
        public string FullName { get; set; }
    }

    public class NewEmployeeCommandHandler : IRequestHandler<NewEmployeeCommand, int>
    {
        private readonly IApplicationDbContext _context;
        private readonly ICurrentUserService _currentUser;

        public NewEmployeeCommandHandler(
            IApplicationDbContext context,
            ICurrentUserService currentUser
        )
        {
            _context = context;
            _currentUser = currentUser;
        }
        public async Task<int> Handle(NewEmployeeCommand request, CancellationToken cancellationToken)
        {
            var employee = new Employee{
                FullName = request.FullName,
                Email = _currentUser.Email
            };

            _context.Employees.Add(employee);

            return _context.SaveChangesAsync();
        }
    }



   // Employee Controller

  [HttpPost]
  public async Task<int> CreateNewEmployee([FromBody] NewEmployeeCommand request){
        return await Mediator.Send(request);
   }
```


This works perfectly fine to me  as you don't have to define this in the command 
```cs 
[JsonIgnore] 
public string Email { get; set; }
```
and also I can't seem to see which abstraction is being abstracted again as what you are talking about. I hope this helps you well. :)

**Reply by ssahinh** [reply]
Thank you, it was pretty self explanatory. :)

how can i mark as answered really i don't know but. your answer is okay for me.
